### PR TITLE
Rewrite installation and upgrade docs in README

### LINKED ISSUE #398: Improve Minor upgrade instructions
Thanks for maintaining this great deck. I'm trying to update from 4.0 to 4.1
When performing a minor upgrade i follow the instructions and use crowd anki.
But im not sure what should be ticked or not. 

In my screenshot is the default selection and the instructions on the right. 

Screenshot 2021-02-14 173709

Secondly there's no mention of a new deck being created. I have "ultimate georgaphy_2" what's going on here?
image

### PR DESCRIPTION
**Pull request by axelboc**
Fix #398. Hopefully it's a lot clearer:

- The _Getting started_ section now explains how to install the deck with CrowdAnki. I no longer mention the APKG install so that people don't end up in a situation where they have a duplicated deck on upgrade. Also, the steps are now the same when installing and upgrading, which is much simpler to understand.
- I document a few common upgrade scenarios in the _Upgrading_, as you'll see. Am I missing any?
  - Should we add scenarios for when people change the templates/styles/fields?
  - Is the _Standard to extended_ section still accurate, or could we provide people with a better solution?
- Should I add a note about importing two decks in two different languages somewhere?

---

**EDIT**

I've added a new section called _Customising the deck_ that goes through a number of changes that users might be interested in making to their local UG decks. It explains how some changes may affect or be reversed by future upgrades, how to go about preserving the changes when possible.

### CODE REVIEW

**Comment by aplaice** [comment]
> Is the Standard to extended section still accurate, or could we provide people with a better solution?

>  Importing an extended deck on top of a standard deck may bring unepected results. Instead, we recommend that you start learning the extended deck from scratch. To do so, either:

Importing an extended deck on top of a standard one may indeed bring unexpected (and without care undesired) results. However, with proper care, it can be done without issues:

1. Start the import the of the extended deck on top of the standard deck, as normal.
2. In the "CrowdAnki Import Settings" dialog, either tick the "Do not move existing cards" box, or not, as normal.
3. A "Change Note Type" dialog will appear, unlike normally. Here, the mapping between the existing Cards and the new Cards and between the existing Fields and the new Fields is defined. 
    Nothing needs to be changed in the "Fields" section, as the order of the Fields is the same in both note types (standard and extended).
    However, in the Cards section, you _**need** to set "Change Flag - Country to: Flag - Country" and "Change Map - Country to: Map - Country" (the "defaults"\* are mapping "Flag - Country" to "Country - Flag" and "Map - Country" to "Flag - Country", which is obviously wrong). 

\* The defaults are this way because of the order of the templates in the two note types. For instance, see the current [`recipes/source_to_anki.yaml`](https://github.com/axelboc/anki-ultimate-geography/blob/master/recipes/source_to_anki.yaml#L50-L67). (It was the same when using `anki-dm`.) Unfortunately, we can't easily rearrange the templates, so that the first four "standard" templates were also the first four "extended" templates, in the same order, because that would result in progress loss (or worse progress mixup) for everybody who was upgrading.

(FWIW I had upgraded from the standard to the extended deck in this way, ~ two years ago.)

File: README.md [hunk 1]
```diff
@@ -87,52 +87,67 @@ Anki's [search feature](https://docs.ankiweb.net/#/searching), coupled with the
...
+
+### First upgrade after APKG import
+
+You may have initially installed _Ultimate Geography_ by importing an **APKG file**. You may have downloaded such a file from this repo or from the [deck's page](https://ankiweb.net/shared/info/2109889812) on AnkiWeb, as this used to be the recommended installation process.
+
+If you're in this situation and wish to upgrade, proceed as follows:
 
-> For more information about versioning and the differences between minor and major, please refer to the [_Versioning_](CONTRIBUTING.md#versioning) section.
+1. Perform the _Getting started_ steps exactly.
+1. You should end up with two decks: your original deck called "Ultimate Geography" and a new, duplicated deck called "Ultimate Geography_2". Every card in your original deck should have moved automatically to the new deck and your original deck should now be empty. You can verify this in Anki's [card browser](https://docs.ankiweb.net/#/browsing?id=browsing).
```
Comment (aplaice) [inline README.md:121]: > your original deck should now be empty

Should we mention something like "unless you created some notes of your own"?

File: README.md [hunk 2]
```diff
@@ -87,52 +87,67 @@ Anki's [search feature](https://docs.ankiweb.net/#/searching), coupled with the
...
+If you've moved some of the cards out of the default _Ultimate Geography_ deck and into another deck, by default CrowdAnki will move those cards back on import.
 
-> If you've moved some of the cards out of the default _Ultimate Geography_ deck and into another deck, by default CrowdAnki will move those cards back on import. This behaviour [can be disabled](https://github.com/Stvad/CrowdAnki#configuration-settings).
+To prevent this behaviour and update the existing cards in place, follow the steps in the [_Getting started_](@getting-started) section but in the last step, make sure to tick the _Do Not Move Existing Cards_ checkbox in the _CrowdAnki Import Settings_ dialog box.
+
+### Standard to extended
+
+Importing an extended deck on top of a standard deck may bring unepected results. Instead, we recommend that you start learning the extended deck from scratch. To do so, either:
```
Comment (aplaice) [inline README.md:132]: Typo: "unepected results" → "unexpected results".

**Review by aplaice — APPROVED** [review]

**Comment by aplaice** [comment]
> Should we add scenarios for when people change the templates/styles/fields?

There's no neat solution for this ([though there are workarounds, at least for the first two](https://github.com/axelboc/anki-ultimate-geography/discussions/400)), so perhaps?

> Should add a note about importing two decks in two different languages somewhere?

I think we should wait with mentioning it explicitly until we've disentangled French from Spanish and Norwegian from English.

> Am I missing any?

Perhaps the user having created their own cards, which obviously causes no issues?

<hr/>

Perhaps we could mention something to the effect of "for unusual upgrades and when you yourself have made changes other than those mentioned there are possible upgrade procedures or workarounds, as described here", linking to a Wiki page (so that anybody could add items there), which would, in turn link to multiple discussion pages, one per topic, similarly to what you had suggested regarding per-upgrade discussion pages.

(A Wiki page to emphasise that they're not "official"/recommended/that while we're willing to help, the user is responsible for whatever happens.)

The tricky standard→extended deck upgrade could also be mentioned in such a wiki and/or discussion page.

File: README.md [hunk 3]
```diff
@@ -129,11 +129,17 @@ To prevent this behaviour and update the existing cards in place, follow the ste
 
 ### Standard to extended
 
-Importing an extended deck on top of a standard deck may bring unepected results. Instead, we recommend that you start learning the extended deck from scratch. To do so, either:
+Importing an extended deck on top of a standard deck may bring unexpected results. You should consider learning the extended deck from scratch. To do so, either:
 
 - remove the standard deck first by following the instructions in the [_Major version_](#major-version) section below, or
 - import the extended deck in a separate Anki profile.
 
+If you'd rather keep your progress, proceed as follows:
```
Comment (aplaice) [inline README.md:137]: Looks great! (Less verbose and more clear than what I wrote!)

**Comment by axelboc** [comment]
I'll add a section for changes to fields, templates, etc. and point to the discussion. I'll keep things "high level" -- we can always point to other discussions later on with more detailed solutions.

> I think we should wait with mentioning it explicitly until we've disentangled French from Spanish and Norwegian from English.

Let me rephrase: should we add a warning to recommend to _not_ attempt importing two languages? 😁 

> Perhaps the user having created their own cards, which obviously causes no issues?

I guess a user's own cards could be affected if we made a radical change to fields or templates... To cover all our bases, I've added "Unless otherwise mentioned in the release notes, if you've added your own cards to the deck, you can be reassured that upgrading will not affect them whatsoever."

---

I like your idea of a wiki page, but I think we can manage without. I don't see a problem of having a few more unusual scenarios in the README (as long as they don't relate to specific languages/releases of the deck) -- if the solutions are too long, we can always point to the relevant issues/discussions. And for very obscure scenarios, I think once-off (but still searchable) issues/discussions are fine.

**Comment by aplaice** [comment]
> Let me rephrase: should we add a warning to recommend to not attempt importing two languages? grin

I don't think it's worth it. It's a relatively niche situation and the README is long already. If somebody does do it, then they'll either have the deck in both languages or have their existing deck translated into the target language (other than the last seven notes added in 4.1, which will be created as new cards). Neither of these is catastrophic and if somebody becomes worried, they can ask here.

Since, once we disentangle EN/NB and FR/ES and upgrade everyone using CZ/NL/RU/SV from 4.1 to 4.1.1, having multiple languages will be explicitly (?) allowed, I'm not sure we want to confuse people by listing all the possible combinations of what can happen when they import two different languages.

We could mention something like "Importing two different languages is currently not explicitly supported, but nothing particularly bad is likely to happen if you do. Please see [here] for details or ask in the discussions section."?

**Comment by axelboc** [comment]
I've spent some time trying all kinds of customisation scenarios, from renaming the deck to adding a new template, and I've documented their impact on upgrading and how people might go about preserving their changes. Let me know what you think!

**Comment by axelboc** [comment]
Sorry for the force-pushed -- hopefully you hadn't started reviewing yet. I'm done now 😄

File: README.md [hunk 4]
```diff
@@ -87,57 +92,74 @@ Anki's [search feature](https://docs.ankiweb.net/#/searching), coupled with the
...
+1. Check that the other cards and fields map correctly and click on _OK_ to start the import.
 
 ### Major version
 
-Upgrading to a major version (e.g. from 2.6 to 3.0) typically leads to a loss of progress. Therefore, unless the [release's page](https://github.com/axelboc/anki-ultimate-geography/releases) tells you otherwise, it is recommended to perform a clean import by following these steps:
+Major versions (e.g. `v3.0`) typically indicate that upgrading may lead to a significant loss of progress, or to the loss of [some of the customisations](#customising-the-deck) you may have made to the deck that would normally be preserved. This can occur, for instance, when [significant changes](https://github.com/axelboc/anki-ultimate-geography/releases/tag/v3.0) are made to the structure of the deck, or when [lots of cards](https://github.com/axelboc/anki-ultimate-geography/releases/tag/v4.0) are removed.
```
Comment (ohare93) [inline README.md:148]:
```suggestion
Major versions (e.g. `v3.0`) typically indicate that upgrading may lead to a significant loss of progress, or to the loss of [some of the customisations](#customising-the-deck) you may have made to the deck that would normally be preserved. This can occur, for instance, when [significant changes](https://github.com/axelboc/anki-ultimate-geography/releases/tag/v3.0) are made to the structure of the deck, or if our include criteria change which can result in [removal of some less relevant cards from the deck.](https://github.com/axelboc/anki-ultimate-geography/releases/tag/v4.0)
```

Read to me like card removal is a common thing that happens all the time! :sweat_smile:

File: README.md [hunk 5]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+## Customising the deck
+
+Interested in making changes to the deck in Anki? Here's what you need to know.
+
+### Changes that are or can be preserved 💖
+
+- **Adding your own cards and media files**
+
+  Go wild. 🌊 Upgrading will not affect any new cards you create, or any new media files you add.
```
Comment (ohare93) [inline README.md:183]: User added media files will not be deleted from their Anki by ugrading, however their inclusion in any of the existing cards in the fields themselves can and probably will be overwritten.

File: README.md [hunk 6]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+  If a duplicated deck is [created on import](#first-upgrade-after-apkg-import), your own cards will stay in their original deck.
+
+- **Moving cards into another deck**
+
+  Some people prefer to combine multiple decks into one to review them together.
+
+  By default, CrowdAnki will move all the cards back into the _Ultimate Geography_ deck on import. To prevent this behaviour and update the existing cards in place, follow the steps in the [_Getting started_](@getting-started) section but in the last step, make sure to tick the _Do Not Move Existing Cards_ checkbox in the _CrowdAnki Import Settings_ dialog box.
```
Comment (ohare93) [inline README.md:191]:
```suggestion
  By default, CrowdAnki will move all the cards back into the _Ultimate Geography_ deck on import. To prevent this behaviour make sure to tick the _Do Not Move Existing Cards_ checkbox in the _CrowdAnki Import Settings_ dialog box to update the existing cards in place, when following the steps in the [_Getting started_](@getting-started) section.
```

Give the pertinent info asap. Helps greatly in these long ReadMe files when people are scanning for the info the want.

File: README.md [hunk 7]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+  - suspending or burying the card;
+  - creating a [filtered deck](#custom-study) that doesn't include the card;
+  - moving the card into an "icebox" deck and ticking the _Do Not Move Existing Cards_ the next time you upgrade, as explained in the previous section, under _Moving cards into another deck_.
+
+### Changes that prevent upgrading ⛔
+
+- **Adding a new field (population, currency, etc.)**
+
+  Adding a new field to the _Ultimate Geography_ note type will prevent you from upgrading the deck entirely. When you'll try to import it again, Anki will likely error and you'll risk losing your entire progress. Hopefully Anki, CrowdAnki, Brain Brew and UG will [one day](https://github.com/ohare93/brain-brew/issues/4#issuecomment-644975261) find a way to make this possible.
```
Comment (ohare93) [inline README.md:250]:
```suggestion
  Adding a new field to the _Ultimate Geography_ note type will prevent you from upgrading the deck entirely. When you'll try to import it again, Anki will likely error and you'll risk losing your entire progress. Hopefully Anki, CrowdAnki, Brain Brew, and UG will [one day](https://github.com/ohare93/brain-brew/issues/4#issuecomment-644975261) find a way to make this possible.
```

Oxford Comma! :wink: 

And god damn, linking people to the mega thread is just mean :rofl: :+1:

File: src/headers/default.yaml [hunk 8]
```diff
@@ -6,7 +6,7 @@ extendRev: 50
 deck_configurations:
 - __type__: DeckConfig
   crowdanki_uuid: 43d37fac-9a65-11e8-9cc5-a0481cc15658
-  name: Default
+  name: Ultimate Geography
```
Comment (ohare93) [inline src/headers/default.yaml:9]: Whoops, how was this still named Default? :sweat_smile:

File: README.md [hunk 9]
```diff
@@ -18,15 +18,20 @@ The deck is available in **English**, **German**, **Spanish**, **French**, **Nor
...
-  - [Minor version](#minor-version)
+  - [First upgrade after APKG import](#first-upgrade-after-apkg-import)
+  - [Keeping deck customisations](#keeping-deck-customisations)
+  - [Levelling up from standard to extended](#levelling-up-from-standard-to-extended)
   - [Major version](#major-version)
 - [**Deck structure**](#deck-structure)
+- [**Customising the deck**](#customising-the-deck)
+  - [Changes that are or can be preserved](#changes-that-are-or-can-be-preserved-)
+  - [Changes that get reverted](#changes-that-get-reverted-)
+  - [Changes that prevent upgrading](#changes-that-prevent-upgrading-)
```
Comment (ohare93) [inline README.md:30]: Useless -s at the end of the link? I guess they are no harm.

**Review by ohare93 — COMMENTED** [review]
Looks great, fantastic work :+1:

File: README.md (hunk 5 above)
Comment (ohare93) [reply README.md:183]: And then if they are not references Anki will remove them on a Media check.

File: README.md [hunk 10]
```diff
@@ -87,57 +92,74 @@ Anki's [search feature](https://docs.ankiweb.net/#/searching), coupled with the
...
+### Keeping deck customisations
 
-> For more information about versioning and the differences between minor and major, please refer to the [_Versioning_](CONTRIBUTING.md#versioning) section.
+If you've made, or are thinking of making, any changes to your _Ultimate Geography_ deck, check the [_Customising the deck_](#customising-the-deck) section below to see what will happen to them the next time you upgrade, and whether you need to take any steps to preserve them.
 
-### Minor version
+### Levelling up from standard to extended
 
-For a minor version upgrade (e.g. from 2.3 to 2.6), _do not_ import the APKG package in Anki as you may have done initially or you will lose your progress. Instead, proceed as described in the [_Other languages and versions_](#other-languages-and-versions) section, making sure to pick the correct archive (e.g. `Ultimate_Geography_v[...]_EN.zip` for the standard English deck).
+Importing an extended deck on top of a standard deck may bring unexpected results. You should consider learning the extended deck from scratch. To do so, either:
```
[the remark this answers is not in the document]
Comment (aplaice) [reply README.md:135]: If the users just click-through all the dialogues (just selecting OK), which many people will do (and I don't blame them — given the general preponderance of confirmations dialogues), then they will get unexpected results (they'll end up having a jumbled-up review schedule (with what are really new cards ("Country - Flag") presented as old cards, and vice-versa, which is arguably far worse than starting from scratch).

You're right, though, that we might wish to soften the language slightly.

Perhaps:

`may bring unexpected results` → `can be tricky and needs care`

and

`You should consider learning the extended deck from scratch` → `You may instead consider learning the extended deck from scratch`

?

File: README.md (hunk 5 above)
Comment (aplaice) [reply README.md:183]: Perhaps:

```suggestion
  Go wild. 🌊 Upgrading will not affect any new cards you create, or any new media files you add to the newly added cards.
```

?

File: README.md (hunk 4 above)
Comment (axelboc) [reply README.md:148]: I'm actually tempted to remove this example completely. After all, we'll never remove as many cards as the last time we did. The seas criteria will only remove a couple of cards, which I don't think will justify a major release.

File: README.md (hunk 6 above)
Comment (axelboc) [reply README.md:191]: Good call, though I think "and update the existing cards in place" needs to stay put for the sentence to remain clear:

> To prevent this behaviour and update the existing cards in place, make sure to tick the _Do Not Move Existing Cards_ checkbox in the _CrowdAnki Import Settings_ dialog box, when following the steps in the [_Getting started_](@getting-started) section.

File: src/headers/default.yaml (hunk 8 above)
Comment (axelboc) [reply src/headers/default.yaml:9]: Yeah, makes it very tricky to distinguish it from Anki's _Default_ option group 😄

File: README.md (hunk 9 above)
Comment (axelboc) [reply README.md:30]: I believe this is how GitHub identifies anchors to headings that end in `[space][emoji]`. I'll fix on `master` if it's not the case after merging.

File: README.md (hunk 4 above)
Comment (axelboc) [reply README.md]: We may remove some of the lakes with seas in their name, but again, not a huge removal.

**Comment by axelboc** [comment]
Latest changes in https://github.com/axelboc/anki-ultimate-geography/pull/403/commits/03ab5aefe756bf9123f01a72cf83454259eb8add

File: README.md [hunk 11]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+  When importing the deck for the first time, CrowdAnki creates a new "option group" called _Ultimate Geography_ (unless it already exists). If you've made any changes to this group's options, they will be reverted on import.
+
+  To avoid losing your options, create your own option group and give it a distinctive name. After you upgrade, you'll just have to switch the deck back to this option group.
+
+- **Customising the templates and styles**
+
+  Any edits you make to the built-in templates of the _Ultimate Geography_ note type will be reverted on upgrade. Same goes with the CSS styles that are shared by all the templates.
+
+  One way to workaround this and preserve your changes is to  first duplicate the note type and switch all the cards to the new note type in the Anki browser. You can then make your template and style changes in the duplicate note type without affecting the original note type. The next time you upgrade, all the cards will switch back to the original note type, but you can easily switch them back to your customised note type in the Anki browser.
```
Comment (aplaice) [inline README.md:207]: Yes, this is far better (and more future-proof) than my approach!

Perhaps we should phrase it as "cloning" the note model, as that's how it's called in the interface as well as in the manual? Also, should we link the manual in case people don't know how to do it (I barely remembered and I've done it before...)?

```suggestion
  One way to workaround this and preserve your changes is to first [clone the note type](https://docs.ankiweb.net/#/editing?id=adding-a-note-type) and switch all the cards to the new note type in the Anki browser. You can then make your template and style changes in the cloned note type without affecting the original note type. The next time you upgrade, all the cards will switch back to the original note type, but you can easily switch them back to your customised note type in the Anki browser.
```

(It's possible that I'm missing an easier way of duplicating note models...)

<hr/>

I've tested it, and it works as desired and expected.

However, I'm slightly worried, as the `crowdanki_uuid` of the "cloned" note model is the same as that of the original one. (I've checked by exporting the deck with the cloned note model and by inspecting `collection.anki2` with sqlite.) (This is analogous to https://github.com/Stvad/CrowdAnki/issues/118, just for note models...) Hence, in principle, `CrowdAnki` could "update" the new (cloned) note model, rather than the original.

Fortunately, it seems (looking at [the UUID code](https://github.com/Stvad/CrowdAnki/blob/a583fa8ceaa4932cf00630476e357ef2e420233e/crowd_anki/utils/uuid.py#L34)) that `CrowdAnki` finds the first matching note_model (the first one having a matching UUID), and assuming that Anki keeps its note_models in chronological order, this will be the original note_model (as we desire), so there _shouldn't_ be problems (but I'm still not 100% comfortable...).

File: README.md (hunk 4 above)
Comment (aplaice) [reply README.md]: We might also end up removing some ("geographical" rather than "political") islands.

Comment (axelboc) [reply README.md]: Ah right. All in all, I still don't think it will amount to "losing a significant amount of progress", especially considering that removed cards are not actually removed from people's decks on import.

I think I'll remove the quote that points to the `CONTRIBUTING` guide's _Versioning_ section. I don't think this section provides any more information than is already in the `README`.

File: README.md [hunk 12]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+  > Note that you may have to update your custom templates and styles afterwards to stay in line with the built-in templates.
+
+- **Adding or removing a template**
+
+  We're getting into dangerous territory here, but you should be able to make this work by duplicating the default note type as explained in the previous point, _Customising the templates and their styles_.
+
+  The difference is that, when you import the deck and/or when you switch back the cards to your customised note type, you may see a dialog box asking you to match the templates of one note type to the templates of the other note type. (This same dialog box appears when importing the extended deck [on top of the standard deck](#levelling-up-from-standard-to-extended).)
+
+  As long as you map the templates that Anki asks you to match correctly, you should be fine. However, we strongly recommend that you try this out in a separate Anki profile first.
```
Comment (aplaice) [inline README.md:217]: tldr; Do not recommend :scream:

I might be doing something wrong, but I don't see how upgrading a deck with an added card template can be done without loss of progress (without manually fiddling with the `deck.json` beforehand (my workaround of first exporting your current deck and then re-importing doesn't really work (or perhaps it works, while crashing Anki??)!)).

With the UUID cloning issue, the note model is (for me) updated "in-place" without any dialogue, unlike in the Standard → Extended case (or unlike if the original model had had an extra card added), since here 

1. the UUIDs of the two note models are the same
2. the original note model is not changed (no cards/fields added/removed)

Hence, all the notes are silently moved back to the "original" note model and silently have one card template removed, (without the corresponding "extra"/now attached cards removed). As a result, when I open the browser, I get a "list index out of range" error (since there are cards attached to the "original" note model which have an index higher than the number of cards/note...) and can't do anything. If I try to import an export of the deck with the cloned note model, then I get a key error (again because there are cards that have no right to exist...) (though after killing Anki and restarting, it seems that all cards including the ones from the added templates are restored, again, with progress). Alternatively, (instead of trying to import an export of the deck with the cloned note) I can check the database, but that deletes the extra cards.

If the UUID cloning issue were not present, (i.e. the cloned deck had its own UUID, rather copying that of the original) then you'd be presented with the standard note model migration dialogue (like for Standard → Extended) and would have to force mapping the added cards (those corresponding to the added template) to "Nothing"...

<hr/>

Removing templates should be fine.

File: README.md (hunk 11 above)
Comment (axelboc) [reply README.md]: Hmm, that's disappointing. That being said, the worst that can happen is that CrowdAnki updates the cloned note type, in which case the user would just lose their customisation. Maybe I can just add a warning that this may not work in the future or something and to recommend backing up the deck before upgrading.

File: README.md [hunk 13]
```diff
@@ -149,3 +171,84 @@ Every note is tagged. The tags, which are listed below by category, are prefixed
...
+
+  Upgrading will bring the card back. If you don't want to learn a card at all or if you want to delay learning it, consider:
+
+  - suspending or burying the card;
+  - creating a [filtered deck](#custom-study) that doesn't include the card;
+  - moving the card into an "icebox" deck and ticking the _Do Not Move Existing Cards_ the next time you upgrade, as explained in the previous section, under _Moving cards into another deck_.
+
+### Changes that prevent upgrading ⛔
+
+- **Adding a new field (population, currency, etc.)**
```
Comment (aplaice) [inline README.md:248]: Ugh, yeah, you're right. This (the fact that you get a `list index out of range` error rather than having the extra field successfully mapped to Nothing) is a `CrowdAnki` bug. :(

File: README.md (hunk 11 above)
Comment (aplaice) [reply README.md]: > this may not work in the future

Hopefully, I'll get around to fixing https://github.com/Stvad/CrowdAnki/issues/118 before sufficient things change in the Anki internals that this stops working.

> recommend backing up the deck before upgrading

FWIW `CrowdAnki` does a backup before every import, but doing one manually won't hurt.

**Review by aplaice — APPROVED** [review]
Summing up, this looks really useful! :)

File: README.md (hunk 12 above)
Comment (axelboc) [reply README.md:217]: You're right, I get the error as well. I think when I tested, I must have added one template and removed another, which led to the same number of templates and may explain why I didn't see the error.

Comment (axelboc) [reply README.md:217]: Removing templates by cloning the note type ends up being more complicated than just removing the template again after the upgrade. So I'll move it into the next section.

**Comment by axelboc** [comment]
The latest batch of changes: https://github.com/axelboc/anki-ultimate-geography/pull/403/commits/b5e3fcd1548f3e9ecffa3c679140b6b81fdc809f
